static/js/stream_settings_ui.js

```diff
@@ -745,10 +745,12 @@ export function change_state(section) {
     // if the section is a valid number.
     if (/\d+/.test(section)) {
         const stream_id = Number.parseInt(section, 10);
-        // Guest users can not access unsubscribed streams
-        // So redirect guest users to 'subscribed' tab
-        // for any unsubscribed stream settings hash
-        if (page_params.is_guest && !stream_data.is_subscribed(stream_id)) {
+        const sub = sub_store.get(stream_id);
+        // We have two cases when settings of streams are not accessible.
+        // 1. Guest users can not access unsubscribed streams.
+        // 2. Stream settings for a deleted stream are being accessed.
+        // In both these case we redirect the user to 'subscribed' tab.
+        if ((page_params.is_guest && !stream_data.is_subscribed(stream_id)) || !sub) {
```

Comment:
Why is `!sub` a correct check for the stream being deleted / not existing?

Reply:
As a further question, do we expect `sub` to be `undefined` for guest users too?

Reply:
Yes, I checked that we do not have the sub data for unsubscribed. But there's a case where a guest user is having stream settings overlay open and is unsubscribed by some admin at the same time. In that case that guest user can go to that stream setting using by using browser back button if we only check `if (!sub)`. Having `(page_params.is_guest && !stream_data.is_subscribed(stream_id))` check ensures that guest users are not able to open stream settings whose data is present (probably because they were subscribed earlier) but are not subscribed now.

One other solution could be that unsubscribe event also deletes the stream data from `sub_store` for guest users.

Reply:
I have updated the comment to mention about the third case as well.